app/wizards/claims/add_claim_wizard/mentor_training_step.rb

```diff
@@ -0,0 +1,64 @@
+class Claims::AddClaimWizard::MentorTrainingStep < BaseStep
+  attribute :hours_completed
+  attribute :mentor_id
+  attribute :custom_hours_completed
+
+  validates :mentor_id, presence: true
+  validates :hours_completed, presence: true
+  validates(
+    :hours_completed,
+    presence: true,
+    numericality: {
+      only_integer: true,
+      greater_than_or_equal_to: 1,
+      less_than_or_equal_to: :max_hours,
+    },
+    unless: :custom_hours_selected?,
+  )
+
+  validates(
+    :custom_hours_completed,
+    presence: true,
+    numericality: { only_integer: true },
+    between: { min: 1, max: :max_hours },
+    if: :custom_hours_selected?,
+  )
```

Comment:
I wonder if it'd simplify things to model it like this:

```ruby
attribute :hours_to_claim, :string
attribute :custom_hours, :integer
```

Where `hours_to_claim` can either be `"maximum"` or `"custom"`.
And `custom_hours` is either `nil` or an integer.

And then you can have a method:

```ruby
def hours_completed
  hours_to_claim == "maximum" ? max_hours : custom_hours
end
```

That way, you'll only need one `numericality` validation rule rather than duplicating it across two fields, and needing conditional based on `custom_hours_selected?`.

Does that make sense? 🤔 I'm not sure how well I've explained it.

Reply:
Whilst a nice concept, I'm not sure how we would implement this with the form.

Reply:
So most of this works, but I can't use `attribute :custom_hours, :integer` because the `govuk_number_field` allows for floats, and `attribute :custom_hours, :integer` converts it immediately to an integer. So `2.5` becomes `2`, with no warning. Which breaks validation.